File: MailVestBack/Mantis.Comun/Constantes/Permisos/Permisos.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace MailVest.Comun.Constantes.Permisos
{
    public static class Permisos
    {

        public static class Configuracion
        {
            public static class AdminUsuarios
            {
                public const string Alta = "UserAlta";
                public const string Baja = "UserBaja";
                public const string Cambio = "UserCambio";
                public const string Exportar = "UserExportar";

                public static readonly string[] Todos =
                {
                    Alta,
                    Baja,
                    Cambio,
                    Exportar,
                };
            }

            public static class AdminRoles
            {
                public const string Alta = "RolAlta";
                public const string Baja = "RolBaja";
                public const string Cambio = "RolCambio";
                public const string Exportar = "RolExportar";
                public static readonly string[] Todos =
                    {
                    Alta,
                    Baja,
                    Cambio,
                    Exportar,
                };
            }
            public static class Catalogos
            {
                public static class AdminParametros
                {
                    public const string Alta = "ParametrosAlta";
                    public const string Baja = "ParametrosBaja";
                    public const string Cambio = "ParametrosCambio";
                    public static readonly string[] Todos = {
                    Alta,
                    Baja,
                    Cambio,
                    };
                }
                public static class AdminAprobadores
                {
                    public const string Alta = "AprobadorAlta";
                    public const string Baja = "AprobadorBaja";
                    public const string Cambio = "AprobadorCambio";
                    public const string CambioDefault = "AprobadorCambDefault";
                    public static readonly string[] Todos =
                    {
                        Alta,
                        Baja,
                        Cambio,
                        CambioDefault
                    };
                }
            }
        }

        public static class Correos
        {
            public static class Objetivos
            {
                public const string Alta = "ObjetivoAlta";
                public const string Cambio = "ObjetivoCambio";
                public const string Baja = "ObjetivoBaja";
                public static readonly string[] Todos =
                {
                    Alta,
                    Baja,
                    Cambio
                };
            }
            public static class Plantillas
            {
                public const string Alta = "PlantillaAlta";
                public const string Cambio = "PlantillaCambio";
                public const string Exportar = "PlantillaExportar";
                public const string Baja = "PlantillaBaja";
                public static readonly string[] Todos =
                {
                    Alta,
                    Baja,
                    Cambio,
                    Exportar,
                };
            }
            public static class CorreosC
            {
                public const string Alta = "CorreoAlta";
                public const string Cambio = "CorreoCambio";
                public const string Exportar = "CorreoExportar";
                public const string Envio = "CorreoEnvio";
                public const string Editar = "CorreoEditar";
                public static readonly string[] Todos =
                {
                    Alta,
                    Cambio,
                    Exportar,
                    Envio,
                    Editar,
                };
            }
        }
    }
}

File: MailVestFront/src/Constantes/Permisos.ts
export const Permisos = {
    Configuracion: {
        codigo: "Configuracion",
        label: "Configuración",
        icon: "pi pi-cog",
        submodulos: {
            AdminUsuarios: {
                codigo: "AdminUsuarios",
                label: "Usuarios",
                icon: "pi pi-user",
                ruta: "/Usuarios",
                permisos: {
                    Acceso: "AdminUsuarios",
                    Alta: "UserAlta",
                    Baja: "UserBaja",
                    Cambio: "UserCambio",
                    Exportar: "UserExportar"
                }
            },
            AdminRoles: {
                codigo: "AdminRoles",
                label: "Roles",
                icon: "pi pi-address-book",
                ruta: "/Roles",
                permisos: {
                    Acceso: "AdminRoles",
                    Alta: "RolAlta",
                    Baja: "RolBaja",
                    Cambio: "RolCambio",
                    Exportar: "RolExportar"
                }
            },
        Catalogos: {
            codigo: "Catalogos",
            label: "Catalogos",
            icon: "fa-solid fa-folder-tree",
            submodulos: {
                AdminParametros: {
                    codigo: "AdminParametros",
                    label: "Parametros",
                    icon: "fa-solid fa-toolbox",
                    ruta: "/Parametros",
                    permisos: {
                        Acceso: "AdminParametros",
                        Alta: "ParametrosAlta",
                        Baja: "ParametrosBaja",
                        Cambio: "ParametrosCambio",
                    }
                },
                AdminAprobador: {
                    codigo: "AdminAprobadores",
                    label: "Aprobadores",
                    icon: "pi pi-users",
                    ruta: "/Aprobadores",
                    permisos: {
                        Acceso: "AdminAprobadores",
                        Alta: "AprobadorAlta",
                        Baja: "AprobadorBaja",
                        Cambio: "AprobadorCambio",
                        CambioDefault: "AprobadorCambDefault",
                    }
                }
            }
        },
        },
    },
    Correos: {
        codigo: "Correos",
        label: "Correos",
        icon: "pi pi-envelope",
        submodulos: {
            AdminObjetivos: {
                codigo: "AdminObjetivos",
                label: "Objetivos",
                icon: "",
                ruta: "/Objetivos",
                permisos: {
                    Acceso: "AdminObjetivos",
                    Alta: "ObjetivoAlta",
                    Baja: "ObjetivoBaja",
                    Cambio: "ObjetivoCambio"
                }
            },
            AdminPlantillas: {
                codigo: "AdminPlantillas",
                label: "Plantillas",
                icon: "fa-solid fa-envelope-open-text",
                ruta: "/Plantillas",
                permisos: {
                    Acceso: "AdminPlantillas",
                    Alta: "PlantillaAlta",
                    Baja: "PlantillaBaja",
                    Cambio: "PlantillaCambio",
                    Exportar: "PlantillaExportar"
                }
            },
            AdminCorreos: {
                codigo: "AdminCorreos",
                label: "Correos",
                icon: "fa-solid fa-paper-plane",
                ruta: "/Correos",
                permisos: {
                    Acceso: "AdminCorreos",
                    Alta: "CorreoAlta",
                    Envio: "CorreoEnvio",
                    Exportar: "CorreoExportar",
                    Cambio: "CorreoCambio",
                }
            },
        }
    },
} as const;